### PR TITLE
refactor: access igraph R package version more easily

### PR DESCRIPTION
**Pull request by maelle**
Related to #1208

I don't understand why the code was so complicated, maybe I'm missing a nuance.

### CODE REVIEW

File: R/aaa-auto.R [hunk 1]
```diff
@@ -3865,3 +3864,15 @@ vertex_path_from_edge_path_impl <- function(graph, start, edge.path, mode=c("out
   res
 }
 
+version_impl <- function() {
+  # Argument checks
+
+
+  on.exit( .Call(R_igraph_finalizer) )
+  # Function call
+  res <- .Call(R_igraph_version, )
```
Comment (maelle) [inline R/aaa-auto.R:3873]: how would I tell Stimulus not to add an empty argument here?

Comment (maelle) [reply R/aaa-auto.R:3873]: I'd appreciate any pointer @Antonov548 @krlmlr (or we can keep this for the hackathon)

Comment (Antonov548) [reply R/aaa-auto.R:3873]: I thought I have fixed this some time ago in stimulus. 
I will take a look

Comment (maelle) [reply R/aaa-auto.R:3873]: that is something I tried to add. I think I got the YAML tweaks wrong!

**Comment by maelle** [comment]
```
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
python3 -m venv .venv
.venv/bin/pip install -r tools/build-requirements.txt
Collecting stimulus
  Cloning https://github.com/igraph/stimulus (to revision 0.21.4) to /tmp/pip-install-zrlobbea/stimulus_1d7f0a76bd484979aeb6c09a9bdb8c5c
  Running command git clone --filter=blob:none --quiet https://github.com/igraph/stimulus /tmp/pip-install-zrlobbea/stimulus_1d7f0a76bd484979aeb6c09a9bdb8c5c
  Running command git checkout -q d7282b21dd4ea880ba01c7b6c6cf72843a644ccc
  Resolved https://github.com/igraph/stimulus to commit d7282b21dd4ea880ba01c7b6c6cf72843a644ccc
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: deepmerge<0.4.0,>=0.3.0 in ./.venv/lib/python3.10/site-packages (from stimulus->-r tools/build-requirements.txt (line 6)) (0.3.0)
Requirement already satisfied: PyYAML<7.0,>=6.0 in ./.venv/lib/python3.10/site-packages (from stimulus->-r tools/build-requirements.txt (line 6)) (6.0.1)
Requirement already satisfied: jinja2<4.0.0,>=3.1.2 in ./.venv/lib/python3.10/site-packages (from stimulus->-r tools/build-requirements.txt (line 6)) (3.1.3)
Requirement already satisfied: MarkupSafe>=2.0 in ./.venv/lib/python3.10/site-packages (from jinja2<4.0.0,>=3.1.2->stimulus->-r tools/build-requirements.txt (line 6)) (2.1.4)
touch .venv/stamp
.venv/bin/stimulus \
           -f src/vendor/cigraph/interfaces/functions.yaml \
           -f tools/stimulus/functions-R.yaml \
           -i tools/stimulus/aaa-auto.R.in \
           -o R/aaa-auto.R \
           -t src/vendor/cigraph/interfaces/types.yaml \
           -t tools/stimulus/types-RR.yaml \
           -l RR
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
fatal: detected dubious ownership in repository at '/root/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /root/workspace
make: *** No rule to make target 'src/interface.c'.  Stop.
```

**Comment by krlmlr** [comment]
What code are you running to get this error message?

**Comment by maelle** [comment]
@krlmlr the docker code, https://github.com/igraph/rigraph/pull/1238/files

**Comment by krlmlr** [comment]
Weird, but confirmed.

Can you please try a plain

```sh
docker run --pull always --rm -ti --platform linux/amd64 -v $(pwd):/root/workspace ghcr.io/cynkra/rig-ubuntu-igraph:main make -f Makefile-cigraph -B
```

in the shell? That worked for me without the errors.

File: R/aaa-auto.R (hunk 1 above)
Comment (maelle) [reply R/aaa-auto.R:3873]: I do not understand why after rerunning the docker command, this problem isn't fixed. This branch has been rebased on main. Is there something else to do for build-requirements to be followed @Antonov548?

File: tools/stimulus/functions-R.yaml [hunk 2]
```diff
@@ -1521,4 +1521,3 @@ igraph_vertex_path_from_edge_path:
 #######################################
 
 igraph_version:
-    IGNORE: RR, RC
```
Comment (maelle) [inline tools/stimulus/functions-R.yaml:1524]: how do I know when to update this and when to update functions.yaml?

**Pull request closed without merging**